docs/sources/variables/variable-types/add-query-variable.md

```diff
@@ -8,6 +8,8 @@ weight = 100
 
 Query variables allow you to write a data source query that can return a list of metric names, tag values, or keys. For example, a query variable might return a list of server names, sensor IDs, or data centers. The variable values change as they dynamically fetch options with a data source query.
 
+Query variables are only supported for strings. To use numbers, convert them to strings and then use them as query variables. For the Azure data source, for example, you can use the [tostring](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/tostringfunction) function for this purpose.
```

Comment:
This depends on the data source, while it's generally the case that only strings are supported, it can be the case that a data source explicitly handles this differently. Also, the link to `tostring` is only valid for Azure Data Explorer. I would say something more generic like:

> Query variables are generally only supported for strings. If your query return numbers or any other data type, you may need to convert them to strings in order to use them as variables.